src/screens/RequestSwap/SelectSwapOption.tsx

```diff
@@ -0,0 +1,138 @@
+
+    const onSelectHandler = (option) => {
+        setSelectedOption(option);
+    };
+
+    const iconDimensions = 50;
+
+    const SwitchSpecifically = () => {
+        return (
+            <TouchableOpacity
+                style={
+                    selectedOption === 0
+                        ? { ...styles.options, ...styles.selected }
+                        : styles.options
+                }
+                onPress={() => onSelectHandler(0)}
+                activeOpacity={1}
+            >
+                <View>
+                    <Text>Switch your duty to a different time</Text>
+                    <View style={{ flexDirection: 'row', justifyContent: 'center' }}>
+                        <Icon
+                            name="person"
+                            height={iconDimensions}
+                            width={iconDimensions}
+                        />
+                        <Icon
+                            name="swap"
+                            height={iconDimensions}
+                            width={iconDimensions}
+                        />
+                        <Icon
+                            name="person"
+                            height={iconDimensions}
+                            width={iconDimensions}
+                        />
+                    </View>
+                </View>
+            </TouchableOpacity>
+        );
+    };
+
+    const SwitchAnyone = () => {
+        return (
+            <TouchableOpacity
+                style={
+                    selectedOption === 1
+                        ? { ...styles.options, ...styles.selected }
+                        : styles.options
+                }
+                activeOpacity={1}
+                onPress={() => onSelectHandler(1)}
+            >
+                <Text>Ask someone to take over this duty</Text>
+                <View style={{ flexDirection: 'row', justifyContent: 'center' }}>
+                    <Icon name="person" height={iconDimensions} width={iconDimensions} />
+                    <Icon
+                        name="arrow-forward"
+                        height={iconDimensions}
+                        width={iconDimensions}
+                    />
+                    <Icon name="person" height={iconDimensions} width={iconDimensions} />
+                </View>
+            </TouchableOpacity>
+        );
+    };
+
+    return (
+        <Layout style={styles.layout}>
+            <TouchableOpacity
+                style={{ position: 'absolute', top: 0, right: 0 }}
+                onPress={props.route.params.closeModal}
+            >
+                <Icon
+                    style={{ width: iconDimensions, height: iconDimensions }}
+                    name="close-square"
+                />
+            </TouchableOpacity>
+
+            <View style={{ flex: 1 }}>
+                <Text>Choose an option</Text>
+            </View>
+            <SwitchSpecifically />
```

Comment:
the code for "switch anyone" and "switch specifically" look really similar. could we reuse them as the same component with props for type "anyone, specific"

Reply:
probably, if I need to reuse this again i'll rewrite it but since its already written, i think we should leave as-is